diff --git a/dist/defaultBlocks-ZzGbYgQn.js b/dist/defaultBlocks-ZzGbYgQn.js
index ebc64e2b6731663da9bddf996a76e60795771763..c3337ee50cbf620bc045411d61f3eee860b471be 100644
--- a/dist/defaultBlocks-ZzGbYgQn.js
+++ b/dist/defaultBlocks-ZzGbYgQn.js
@@ -2718,7 +2718,8 @@ class _n {
         throw new Error("Attempting to update uninitialized suggestions menu");
       n(a, {
         ...this.state,
-        ignoreQueryLength: (s = this.pluginState) == null ? void 0 : s.ignoreQueryLength
+        ignoreQueryLength: (s = this.pluginState) == null ? void 0 : s.ignoreQueryLength,
+        composing: this.pluginState == null ? void 0 : this.pluginState.composing
       });
     }, this.rootEl = o.root, (r = this.rootEl) == null || r.addEventListener("scroll", this.handleScroll, !0);
   }
@@ -2799,6 +2800,7 @@ const B = new Xe("SuggestionMenuPlugin"), Vn = y(({ editor: e }) => {
             if (r.selection.$from.parent.type.spec.code)
               return a;
             const c = r.getMeta(B);
+            const f = e.prosemirrorView.composing;
             if (typeof c == "object" && c !== null) {
               a && (n == null || n.closeMenu());
               const u = Dn(
@@ -2813,14 +2815,15 @@ const B = new Xe("SuggestionMenuPlugin"), Vn = y(({ editor: e }) => {
                 queryStartPos: () => u() + c.triggerCharacter.length,
                 query: "",
                 decorationId: `id_${Math.floor(Math.random() * 4294967295)}`,
-                ignoreQueryLength: c == null ? void 0 : c.ignoreQueryLength
+                ignoreQueryLength: c == null ? void 0 : c.ignoreQueryLength,
+                composing: f
               };
             }
             if (a === void 0)
               return a;
             if (
               // Highlighting text should hide the menu.
-              i.selection.from !== i.selection.to || // Transactions with plugin metadata should hide the menu.
+              !f && !a.composing && i.selection.from !== i.selection.to || // Transactions with plugin metadata should hide the menu.
               c === null || // Certain mouse events should hide the menu.
               // TODO: Change to global mousedown listener.
               r.getMeta("focus") || r.getMeta("blur") || r.getMeta("pointer") || // Moving the caret before the character which triggered the menu should hide it.
@@ -2830,7 +2833,10 @@ const B = new Xe("SuggestionMenuPlugin"), Vn = y(({ editor: e }) => {
               )
             )
               return;
+            if (f)
+              return { ...a, composing: f };
             const l = { ...a };
+            l.composing = f;
             return l.query = i.doc.textBetween(
               a.queryStartPos(),
               i.selection.from
diff --git a/src/extensions/SuggestionMenu/SuggestionMenu.ts b/src/extensions/SuggestionMenu/SuggestionMenu.ts
index 4809607e6e1efddf89ea91992245b23759b9229c..48355143f86621b4a95cff9466c4e341344fe6b0 100644
--- a/src/extensions/SuggestionMenu/SuggestionMenu.ts
+++ b/src/extensions/SuggestionMenu/SuggestionMenu.ts
@@ -15,6 +15,7 @@ const findBlock = findParentNode((node) => node.type.name === "blockContainer");
 export type SuggestionMenuState = UiElementPosition & {
   query: string;
   ignoreQueryLength?: boolean;
+  composing?: boolean;
 };
 
 class SuggestionMenuView {
@@ -38,6 +39,7 @@ class SuggestionMenuView {
       emitUpdate(menuName, {
         ...this.state,
         ignoreQueryLength: this.pluginState?.ignoreQueryLength,
+        composing: this.pluginState?.composing,
       });
     };
 
@@ -146,6 +148,7 @@ type SuggestionPluginState =
       query: string;
       decorationId: string;
       ignoreQueryLength?: boolean;
+      composing?: boolean;
     }
   | undefined;
 
@@ -260,6 +263,7 @@ export const SuggestionMenu = createExtension(({ editor }) => {
               deleteTriggerCharacter?: boolean;
               ignoreQueryLength?: boolean;
             } | null = transaction.getMeta(suggestionMenuPluginKey);
+            const composing = editor.prosemirrorView.composing;
 
             if (
               typeof suggestionPluginTransactionMeta === "object" &&
@@ -289,6 +293,7 @@ export const SuggestionMenu = createExtension(({ editor }) => {
                 decorationId: `id_${Math.floor(Math.random() * 0xffffffff)}`,
                 ignoreQueryLength:
                   suggestionPluginTransactionMeta?.ignoreQueryLength,
+                composing,
               };
             }
 
@@ -300,7 +305,9 @@ export const SuggestionMenu = createExtension(({ editor }) => {
             // Checks if the menu should be hidden.
             if (
               // Highlighting text should hide the menu.
-              newState.selection.from !== newState.selection.to ||
+              (!composing &&
+                !prev.composing &&
+                newState.selection.from !== newState.selection.to) ||
               // Transactions with plugin metadata should hide the menu.
               suggestionPluginTransactionMeta === null ||
               // Certain mouse events should hide the menu.
@@ -319,7 +326,15 @@ export const SuggestionMenu = createExtension(({ editor }) => {
               return undefined;
             }
 
+            if (composing) {
+              return {
+                ...prev,
+                composing,
+              };
+            }
+
             const next = { ...prev };
+            next.composing = composing;
 
             // Updates the current query.
             next.query = newState.doc.textBetween(
diff --git a/types/src/extensions/SuggestionMenu/SuggestionMenu.d.ts b/types/src/extensions/SuggestionMenu/SuggestionMenu.d.ts
index 118b956b4eedd800cacb21eaf16f7addc5cf69de..1dd49b0cf06dd403357bd13497ec39a38c6b3e7b 100644
--- a/types/src/extensions/SuggestionMenu/SuggestionMenu.d.ts
+++ b/types/src/extensions/SuggestionMenu/SuggestionMenu.d.ts
@@ -3,6 +3,7 @@ import { UiElementPosition } from "../../extensions-shared/UiElementPosition.js"
 export type SuggestionMenuState = UiElementPosition & {
     query: string;
     ignoreQueryLength?: boolean;
+    composing?: boolean;
 };
 type SuggestionPluginState = {
     triggerCharacter: string;
@@ -11,6 +12,7 @@ type SuggestionPluginState = {
     query: string;
     decorationId: string;
     ignoreQueryLength?: boolean;
+    composing?: boolean;
 } | undefined;
 export type SuggestionMenuOptions = {
     triggerCharacter: string;
@@ -36,16 +38,19 @@ export declare const SuggestionMenu: (options?: any) => import("../../index.js")
     readonly store: import("@tanstack/store").Store<(UiElementPosition & {
         query: string;
         ignoreQueryLength?: boolean;
+        composing?: boolean;
     } & {
         triggerCharacter: string;
     }) | undefined, (cb: (UiElementPosition & {
         query: string;
         ignoreQueryLength?: boolean;
+        composing?: boolean;
     } & {
         triggerCharacter: string;
     }) | undefined) => (UiElementPosition & {
         query: string;
         ignoreQueryLength?: boolean;
+        composing?: boolean;
     } & {
         triggerCharacter: string;
     }) | undefined>;
